x-pack/legacy/plugins/lens/public/index.ts

```diff
@@ -12,6 +12,7 @@ import 'uiExports/fieldFormats';
 import 'uiExports/search';
 import 'uiExports/visRequestHandlers';
 import 'uiExports/visResponseHandlers';
+import 'uiExports/interpreter';
```

Comment:
Yes, it is needed to register `esaggs` and all other functions and `visualization` renderer. This is needed because otherwise, for example, `esaggs` function is missing. My understanding is that all those should have used `interpreter` ui-export all along.

Why it worked before? I don't know, it used to register `esaggs` function and `visualization` renderer from the same file where `getInterpreter` was imported, my guess is that those were simply registered as a side-effect when `getInterpreter` was imported.

Reply:
My understanding is that `esaggs` and `visualization` functions and `visualizations` renderer where registered as a side-effect when somebody imported `getInterpreter()` function. But they should have used the `interpreter` ui-export&mdash;so it was a bug. In any case, now as nobody is importing the `getInterpreter()` function anymore (except Canvas) those registrants are missing. My understanding is that they should use `interpreter` ui-export. But 🤷‍♂️. Maybe @ppisljar has ideas?

> I'm not sure whether this is needed here... any time you navigate between apps that have import `'uiExports/interpreter'`, won't it re-register all of the legacy functions that plugins have added via `uiExport.interpreter: ['path/to/whatever']` ?

@lukeelmers I don't think I really understand what you mean.

> As an alternative, I think we might be able to get rid of the interpreter uiExport entirely if legacy plugins loaded their "boot" files which register functions as a `hack` instead.

`hacks` ui-export might work. But it is still an ui-export, I'm not sure what benefits it would give use; I see only drawbacks in using it.

Reply:
@streamich what you are saying is my understanding too -- I think I misunderstood your meaning when you said:
> all those should have used interpreter ui-export all along.

I took this to mean "the plugins should have been importing uiExports/interpreter", but what you really meant was "the plugins should have been registering their functions via uiExports.interpreter in the plugin definition". I think we are on the same page now 😃 

> `hacks` ui-export might work. But it is still an ui-export, I'm not sure what benefits it would give us; I see only drawbacks.

I don't feel strongly that using hacks is something we should do here, I only mentioned it because I think it would address @wylieconlon's initial concern which was why we'd need to add these imports to each plugin during migration.

A hack would be loaded everywhere by default, so instead of each legacy app needing to do two things (add an interpreter `uiExport`, and call `import` to get everything back), they could instead do one thing (just add one `hack` and everything else would already be there).

Anyway, this is more of a question of ergonomics and I don't think it should block us from migrating at this point... technically it should work either way. Primarily I wanted to make sure there was a clear answer to Wylie's question, which I think we have come around to now

Reply:
>  won't it re-register all of the legacy functions that plugins have added via uiExport.interpreter: ['path/to/whatever']?
>> I don't think I really understand what you mean.

BTW I was wrong about this ^

I forgot about the fact that each app load is a page refresh, so there isn't any re-registering that happens.